src/tests/dbus-tests/test_bcache.py

```diff
@@ -0,0 +1,132 @@
+import os
+import re
+import time
+
+import storagedtestcase
+
+
+BLOCK_SIZE = 512
+
+
+class StoragedBcacheTest(storagedtestcase.StoragedTestCase):
+    '''This is a basic bcache test suite'''
+
+    @classmethod
+    def setUpClass(cls):
+        storagedtestcase.StoragedTestCase.setUpClass()
+        cls.ensure_modules_loaded()
+
+    def _wipe(self, device):
+        for _ in range(10):
+            ret, _out = self.run_command('wipefs -a %s' % device)
+            if ret == 0:
+                break
+            time.sleep(1)
+        else:
+            self.fail('Failed to wipe device %s after removing bcache.' % device)
+
+    def _force_remove(self, bcache_name, backing_dev, cache_dev):
+        _ret, out = self.run_command('bcache-super-show %s | grep cset.uuid ' % cache_dev)
+        cset_uuid = out.split()[-1]
+
+        self.write_file('/sys/block/%s/bcache/detach' % bcache_name, cset_uuid, True)
+        self.write_file('/sys/fs/bcache/%s/stop' % cset_uuid, '1', True)
+        self.write_file('/sys/block/%s/bcache/stop' % bcache_name, '1', True)
+
+        self._wipe(cache_dev)
+        self._wipe(backing_dev)
+
+    def _get_mode(self, bcache_name):
+        # cache_mode contains all cache modes, the right one is in square brackets
+        modes = self.read_file('/sys/block/%s/bcache/cache_mode' % bcache_name)
+        return next((x[1:-1] for x in modes.split() if re.match(r'^\[.*\]$', x)), None)
+
+    def _get_size(self, bcache_name):
+        cache_dir = '/sys/block/%s/bcache/cache' % bcache_name
+
+        # sum sizes from all caches
+        caches = ['%s/%s' % (cache_dir, d) for d in os.listdir(cache_dir) if re.match('cache[0-9]*$', d)]
+        size = sum([int(self.read_file(os.path.realpath(c) + '/../size')) for c in caches])
+
+        return size
+
+    def test_create_destroy(self):
+        '''Test creating a new bcache and its properties'''
+
```

Comment:
Please make sure the modules are loaded here (``StoragedTestCase.ensure_modules_loaded``)